### PR TITLE
First version for decoder with Inter support

### PR DESCRIPTION
**Pull request by nijilk**

### CODE REVIEW

**Comment by nijilk** [comment]
@joelsole @hassount 
Hi,
Can you please review the changes?

File: CHANGELOG.md [hunk 1]
```diff
@@ -2,12 +2,11 @@
 
 ## [upcoming]
 
-- Inital decoder implementation
+- Decoder Inter support
 - Static library support
 - Alt-ref pictures - temporal filtering
 - Adaptive Transform Block for INTRA
 - Adaptive QP scaling
-- Decoder - Support for Tiles and 10bit
```
Comment (hassount) [inline CHANGELOG.md]: not supported any more ?

File: Docs/svt-av1_decoder_user_guide.md [hunk 2]
```diff
@@ -16,7 +16,9 @@ This document describes how to use the Scalable Video Technology for AV1 Decoder
 
 Initial version:
 
-- Supports only I-frames without loopfilter (deblocking, cdef and self guided restoration filters not supported)
+- Supports all frames without loopfilter (deblocking, cdef and self guided restoration filters not supported)
+- Screen content tolls are not supported
```
Comment (hassount) [inline Docs/svt-av1_decoder_user_guide.md]: typo: tolls -- > tools

File: Source/Lib/Common/Codec/EbInterPrediction.c [hunk 3]
```diff
@@ -56,12 +62,93 @@
 #define ROUND0_BITS 3
 #define COMPOUND_ROUND1_BITS 7
 
-typedef struct SubpelParams {
-    int32_t xs;
-    int32_t ys;
-    int32_t subpel_x;
-    int32_t subpel_y;
-} SubpelParams;
+/* TODO: Add scaling of reference frame support later
```
Comment (hassount) [inline Source/Lib/Common/Codec/EbInterPrediction.c]: do we need this commented out code now ?

File: Source/Lib/Common/Codec/EbInterPrediction.c [hunk 4]
```diff
@@ -987,6 +1074,83 @@ static void highbd_convolve_2d_for_intrabc(const uint16_t *src, int src_stride,
...
+
+    InterpFilterParams filter_params_x, filter_params_y;
+    /* TODO: Add suppport for scaling later */
+    const int32_t is_scaled = 0; // has_scale(subpel_params->xs, subpel_params->ys);
+
+    av1_get_convolve_filter_params(interp_filters, &filter_params_x,
+        &filter_params_y, w, h);
+
+    assert(conv_params->do_average == 0 || conv_params->do_average == 1);
+    //assert(sf);
```
Comment (hassount) [inline Source/Lib/Common/Codec/EbInterPrediction.c]: too much commented out code

File: CHANGELOG.md (hunk 1 above)
Comment (nijilk) [reply CHANGELOG.md]: @hassount 
Tile and 10 bit support is already present.
That's why I moved it from "upcoming" features list.

Comment (nijilk) [reply CHANGELOG.md]: @hassount 
Tile and 10 bit support is already present.
That's why I removed it from "upcoming" features list.

File: Docs/svt-av1_decoder_user_guide.md (hunk 2 above)
Comment (nijilk) [reply Docs/svt-av1_decoder_user_guide.md]: Thanks.
Will fix it.

File: Source/Lib/Common/Codec/EbInterPrediction.c (hunk 3 above)
Comment (nijilk) [reply Source/Lib/Common/Codec/EbInterPrediction.c]: @hassount 
This is for an upcoming normative tool (scaling of reference frame).
This code will help the development when we add the support for this tool.

File: Source/Lib/Common/Codec/EbInterPrediction.c (hunk 4 above)
Comment (nijilk) [reply Source/Lib/Common/Codec/EbInterPrediction.c]: @hassount 
This is for an upcoming normative tool (scaling of reference frame).
This code will help the development when we add the support for this tool.

File: CHANGELOG.md (hunk 1 above)
Comment (hassount) [reply CHANGELOG.md]: I believe it should still be in the change log to be included as part of the changes to v0.6.0

File: Source/Lib/Common/Codec/EbInterPrediction.c (hunk 3 above)
Comment (hassount) [reply Source/Lib/Common/Codec/EbInterPrediction.c]: Then having a debug #define similar to ENTROPY_TRACE_FILE_BASED should be the way to include it I believe

File: CHANGELOG.md (hunk 1 above)
Comment (nijilk) [reply CHANGELOG.md]: @hassount 
Sure. I will add them.

File: Source/Lib/Common/Codec/EbInterPrediction.c (hunk 3 above)
Comment (nijilk) [reply Source/Lib/Common/Codec/EbInterPrediction.c]: @hassount 
Sure.
We will keep them under a MACRO.
Thanks for the review.

**Comment by nijilk** [comment]
> 
> 
> There is a lot of function duplication with the encoder: find_warp_samples, has_overlappable_cand, is_motion_mode_allowed, is_comp_ref_allowed, av1_is_interp_needed,...
> 
> What is the plan for that?

@joelsole 
The functions are replicated because of some of the incompatible data structure.
Few of them can be removed once the harmonization is over.

**Comment by joelsole** [comment]
The build is showing warnings. Could you rebase to see if they are all gone?

**Comment by nijilk** [comment]
> 
> 
> The build is showing warnings. Could you rebase to see if they are all gone?

@joelsole 
I will rebase and check these issues.
I will take other review comments as well.

**Comment by joelsole** [comment]
Superseded by #429, which is a rebased version and not style issue in the commit message.

**Pull request closed without merging**